model-optimizer/mo/ops/slice.py

```diff
@@ -13,133 +13,181 @@
+    """
+    op = 'AttributedSlice'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'type': None,
+            'op': self.op,
+            'in_ports_count': 1,
+            'out_ports_count': 1,
+            'infer': None,
+        }, attrs)
+
+    def supported_attrs(self):
+        return ['axes', 'starts', 'ends']
+
+
+class CaffeSlice(Op):
+    """
+    Slice in Caffe is equivalent to Split operation in OpenVINO.
+    https://caffe.berkeleyvision.org/tutorial/layers/slice.html
+    The operation is replaced with the OpenVINO Split operation on the front phase.
+    """
+    op = 'CaffeSlice'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'type': None,
+            'op': self.op,
+            'in_ports_count': 1,
+            'out_ports_count': 1,
+            'infer': None,
+        }, attrs)
+
+    def supported_attrs(self):
+        return ['slice_point', 'axis']
+
+
+class TFSlice(Op):
+    """
+    Slice operation in Tensorflow is different from Slice in ONNX, Caffe and MXNet. It has begin and size inputs while
+    ONNX Slice and internal MO Slice has start, end, step and axis parameters specified as inputs.
+    https://www.tensorflow.org/api_docs/python/tf/slice
+    The operation is replaced with the internal Slice on the front phase.
+    If size[i] == -1 is replaced to int32_max value for the end.
+    """
+    op = 'TFSlice'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'type': None,
+            'op': self.op,
+            'in_ports_count': 3,
+            'out_ports_count': 1,
+            'infer': None,
+        }, attrs)
+
+
+class MXSlice(Op):
+    """
+    Slice operation in MXNet is different from ONNX, Caffe, Tensorflow. It has begin, end & step attributes
+    https://mxnet.apache.org/versions/1.6/api/python/docs/api/symbol/op/index.html#mxnet.symbol.op.slice
+    The operation is replaced with the OpenVINO StridedSlice operation on the front phase.
+    """
+    op = 'MXSlice'
+    enabled = False
+
+    def __init__(self, graph: Graph, attrs: dict):
+        super().__init__(graph, {
+            'kind': 'op',
+            'type': None,
+            'op': self.op,
+            'in_ports_count': 1,
+            'out_ports_count': 1,
+            'infer': None
+        }, attrs)
 
 
 class Slice(Op):
+    """
+    Semantic of MO internal Slice operation is identical to Slice in ONNX opset >= 10.
+    It has starts, ends, steps and axes inputs.
+    The operation is internal (not present in the OpenVINO opset) and is replaced to StridedSlice.
+    """
     op = 'Slice'
-    enabled = True
+    enabled = False
 
-    def __init__(self, graph: Graph, attrs: dict):
+    def __init__(self, graph: Graph, attrs: dict = None):
```

Comment:
Here you set arrts to None, but everywhere else -- we don't do this. What was original reason to do so?

Reply:
I added this default value to avoid specifying when I call constrictor, e.g. here TFSliceToSlice.py:41
`slice_node = Slice(graph).create_node()`

At the moment it looks like it's the only place where Slice constructor is called explicitly. I guess in the draft versions of PR i was using it more often and it remained here for historical reason. I advocate to leave this default argument here.